### DISCUSSION TITLE
How Do You Position Popover-like Components Relative to the Viewport Without Using Portal?

### DISCUSSION THREAD

**Discussion by elson-currentcorp** [chakra-ui/ark#2584] [Q&A] [answered]
I'm realizing that a lot of focus related issues relating from nesting popover-like components (etc. dialog, select...) can be avoided by only ensuring the topmost component is portalled.

However the downside of this is that if the child popover does not have the portalled attribute, it becomes constrained within the dimensions of the parent popover. 

How can I override the positioner style that the child popover uses so that it is positioned relative to the entire viewport without using the portalled attribute?

I noticed that the <Portal/> component has a container prop. Can we have something like that for PopoverLikeComponent.Positioner as well?

**Comment by segunadebayo** [comment, marked as answer]
There are two options:

- Using `position:fixed` on the content
- Setting `positioning.fixed` to `true` to the root
- Using portal but setting the portal container to the root portal content
